services/management/rds_discovery.go

```diff
@@ -0,0 +1,170 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package management
+
+import (
+	"context"
+	"net/http"
+	"sort"
+	"time"
+
+	"github.com/AlekSi/pointer"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/kr/pretty"
+	"github.com/percona/pmm-managed/models"
+	"github.com/percona/pmm-managed/utils/logger"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gopkg.in/reform.v1"
+)
+
+const (
+	// maximum time for AWS discover APIs calls
+	awsDiscoverTimeout = 7 * time.Second
+)
+
+// RDSService RDS Management Service.
+type RDSService struct {
+	db       *reform.DB
+	registry agentsRegistry
+}
+
+// InstanceID uniquely identifies RDS instance.
+// http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Overview.DBInstance.html
+// Each DB instance has a DB instance identifier. This customer-supplied name uniquely identifies the DB instance when interacting
+// with the Amazon RDS API and AWS CLI commands. The DB instance identifier must be unique for that customer in an AWS Region.
+type InstanceID struct {
+	Region string
+	Name   string // DBInstanceIdentifier
+}
+
+type Instance struct {
+	Node    models.Node
+	Service models.Service
+}
+
+// NewRDSService creates new RDS Management Service.
+func NewRDSService(db *reform.DB, registry agentsRegistry) *RDSService {
+	return &RDSService{
+		db:       db,
+		registry: registry,
+	}
+}
+
+func (svc *RDSService) Discover(ctx context.Context, accessKey, secretKey string) ([]Instance, error) {
```

Comment:
That should be an gRPC Server implementing this API: https://github.com/percona/pmm/pull/455